frontend/workflows/projectSelector/src/project-selector.tsx

```diff
@@ -112,36 +153,34 @@ const hydrateProjects = (state: State, dispatch: React.Dispatch<Action>) => {
   }
 };
 
-const ProjectSelector = () => {
+const ProjectSelector = ({ onError }: ProjectSelectorProps) => {
   // On load, we'll request a list of owned projects and their upstreams and downstreams from the API.
   // The API will contain information about the relationships between projects and upstreams and downstreams.
   // By default, the owned projects will be checked and others will be unchecked.
 
   const [customProject, setCustomProject] = React.useState("");
   const { updateSelected } = useDashUpdater();
-  // TODO: restore usage of loadStoredState once we fix the issue with deprecated projects
-  const [state, dispatch] = React.useReducer(selectorReducer, initialState);
+  const [state, dispatch] = React.useReducer(selectorReducer, loadStoredState(initialState));
 
-  React.useEffect(() => {
-    const interval = setInterval(() => hydrateProjects(state, dispatch), 30000);
-    return () => clearInterval(interval);
-  }, []);
-
-  React.useEffect(() => {
-    hydrateProjects(state, dispatch);
-  }, [state[Group.PROJECTS]]);
+  React.useEffect(() => hydrateProjects(state, dispatch), [state[Group.PROJECTS]]);
 
   // computes the final state for rendering across other components
   // (ie. filters out upstream/downstreams that are "hidden")
   const derivedState = React.useMemo(() => deriveStateData(state), [state]);
 
   // This hook updates the global dash state based on the currently selected projects for cards to consume (including upstreams and downstreams).
   React.useEffect(() => {
-    if (!allPresent(state)) {
+    if (!allPresent(state, dispatch)) {
       // Need to wait for the data.
       return;
     }
 
+    const interval = setInterval(() => hydrateProjects(state, dispatch), 30000);
```

Comment:
Moved the setInterval inside of our useEffect that reloads on state changes as I was seeing issues with the same state being sent every time the interval triggered. 

Think this should clearInterval on a re-render/state change and set a new interval to hydrate everything (testing worked at least).

Reply:
By moving this out it will be called every time there is any state change to the variable above. I think the way to handle this would be to continue using the useEffect hook but update the interval every time only the state changes.

Reply:
Ah ok, thought it might've been cleaner having it inside the other useEffect but probably best to keep them separated. I will move it back into the original useEffect and set the dependency on the state.